lib/web/sessions.go

```diff
@@ -342,19 +393,26 @@ func newSessionCache(config *sessionCache) (*sessionCache, error) {
 }
 
 // sessionCache handles web session authentication,
-// and holds in memory contexts associated with each session
+// and holds in-memory contexts associated with each session
 type sessionCache struct {
-	log logrus.FieldLogger
-	sync.Mutex
+	log         logrus.FieldLogger
 	proxyClient auth.ClientI
-	contexts    *ttlmap.TTLMap
 	authServers []utils.NetAddr
+	accessPoint auth.ReadAccessPoint
 	closer      *utils.CloseBroadcaster
 	clusterName string
 	clock       clockwork.Clock
-
 	// cipherSuites is the list of supported TLS cipher suites.
 	cipherSuites []uint16
+
+	mu sync.Mutex
+	// sessions maps user/sessionID to an active web session.
+	// This is the client-facing session handle
+	sessions map[string]*SessionContext
+	// contexts maps user to an active user web session.
+	// These are used to maintain session state that would otherwise
+	// not survive renewal of the session
+	contexts map[string]*sessionContext
```

Comment:
Sorry just looking at this is a little confusing - why are there two "session context" structs, one exported and one not, and what's the difference b/w them? And the fact that one field is called "sessions" and another one "contexts" is also a bit confusing.

Reply:
It is - I'll update to reflect the actual purpose of the unexpected context.